common/app/views/support/ImageProfile.scala

```diff
@@ -137,116 +142,91 @@ class ShareImage(overlayUrlParam: String, shouldIncludeOverlay: Boolean) extends
   override def resizeString: String = {
     if(shouldIncludeOverlay) {
       val params = Seq(widthParam, heightParam, qualityparam, autoParam, fitParam, dprParam, overlayAlignParam, overlayWidthParam, overlayUrlParam).filter(_.nonEmpty).mkString("&")
-      s"?$params"
+
+      if (shouldUpscale) s"?$params&enable=upscale" else s"?$params"
     } else {
       super.resizeString
     }
   }
 }
 
+sealed trait ShareImageCategory
+case object GuardianDefault extends ShareImageCategory
+case object ObserverDefault extends ShareImageCategory
+case object ObserverOpinion extends ShareImageCategory
+case object GuardianOpinion extends ShareImageCategory
+case object Live extends ShareImageCategory
+case class ObserverOldContent(publicationYear: Int) extends ShareImageCategory
+case class GuardianOldContent(publicationYear: Int) extends ShareImageCategory
+case class ObserverStarRating(rating: Int) extends ShareImageCategory
+case class GuardianStarRating(rating: Int) extends ShareImageCategory
+case object Paid extends ShareImageCategory
+
 trait OverlayBase64 {
   def overlayUrlBase64(overlay: String): String =
     Base64.getUrlEncoder.encodeToString(s"/img/static/overlays/$overlay".getBytes).replace("=", "")
 }
 
-object TwitterImage extends OverlayBase64 {
```

Comment:
I have consolidated TwitterImage and FacebookOpenGraphImage into just one - OpenGraphImage - because, as far as I can tell, they are identical(!) other than the boolean used to flag if an overlay should be added to the images, the `TwitterShareImageLogoOverlay.isSwitchedOn` lines and fb equivalents. So I've just parameterised this and removed a ton of code.